sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/Paginator.java

```diff
@@ -128,6 +134,18 @@
                 isChangeFeed,
                 top,
                 maxPageSize),
-                Queues.XS_BUFFER_SIZE);
+                preFetchCount);
+    }
+
+    public static int getPreFetchCount(CosmosQueryRequestOptions queryOptions, int top, int maxPageSize) {
+        int maxBufferedItemCount = queryOptions != null ? queryOptions.getMaxBufferedItemCount() : 0;
```

Comment:
On a 16 cores machine, if the customer configured pagesize <= 6K, then potentially we are going to fetch here will be less than the one in the query pipeline (as It also considers the total cores of the machine).

(do not think it will be a concern, just want to bring it up)

Reply:
Understood - but don't think it makes sense to add the CPU cores and number of partitions here - that will still be prefetched there.